src/webviews/mongoClusters/collectionView/collectionViewRouter.ts

```diff
@@ -10,29 +10,31 @@ import { type MongoClustersClient } from '../../../mongoClusters/MongoClustersCl
 import { MongoClustersSession } from '../../../mongoClusters/MongoClusterSession';
 import { getConfirmationAsInSettings } from '../../../utils/dialogs/getConfirmation';
 import { getKnownFields, type FieldEntry } from '../../../utils/json/mongo/autocomplete/getKnownFields';
-import { publicProcedure, router } from '../../api/extension-server/trpc';
+import { publicProcedure, router, trpcToTelemetry } from '../../api/extension-server/trpc';
 
 import { type CollectionItem } from '../../../mongoClusters/tree/CollectionItem';
 import { showConfirmationAsInSettings } from '../../../utils/dialogs/showConfirmation';
 // eslint-disable-next-line import/no-internal-modules
 import basicFindQuerySchema from '../../../utils/json/mongo/autocomplete/basicMongoFindFilterSchema.json';
 import { generateMongoFindJsonSchema } from '../../../utils/json/mongo/autocomplete/generateMongoFindJsonSchema';
 import { localize } from '../../../utils/localize';
+import { type BaseRouterContext } from '../../api/configuration/appRouter';
 
-export type RouterContext = {
+export type RouterContext = BaseRouterContext & {
     sessionId: string;
     databaseName: string;
     collectionName: string;
     collectionTreeItem: CollectionItem; // needed to execute commands on the collection as the tree APIv2 doesn't support id-based search for tree items.
 };
 
 export const collectionsViewRouter = router({
-    getInfo: publicProcedure.query(({ ctx }) => {
+    getInfo: publicProcedure.use(trpcToTelemetry).query(({ ctx }) => {
```

Comment:
Is it possible to assign `use` for all router paths once? Express allows this
```
const app = new Route()
app.use() // works for all paths in app route
```

Reply:
Interesting. I haven’t seen it so far, but I wasn’t specifically looking for it either. I’ll investigate it if we choose tRPC for the long(er) run.